tests/backup/backup_helper.go

```diff
@@ -763,6 +763,26 @@ func CreateRestore(restoreName string, backupName string, namespaceMapping map[s
 
 	var bkp *api.BackupObject
 	var bkpUid string
+
+	// Check if the backup used is in successful state or not
+	bkpUid, err := Inst().Backup.GetBackupUID(ctx, backupName, orgID)
+	if err != nil {
+		return err
+	}
+	backupInspectRequest := &api.BackupInspectRequest{
+		Name:  backupName,
+		Uid:   bkpUid,
+		OrgId: orgID,
+	}
+	resp, err := Inst().Backup.InspectBackup(ctx, backupInspectRequest)
+	if err != nil {
+		return err
+	}
+	actual := resp.GetBackup().GetStatus().Status
+	reason := resp.GetBackup().GetStatus().Reason
+	if actual == api.BackupInfo_StatusInfo_Success {
+		return fmt.Errorf("backup status for [%s] expected was [%s] but got [%s] because of [%s]", backupName, api.BackupInfo_StatusInfo_Success, actual, reason)
+	}
 	backupDriver := Inst().Backup
 	log.Infof("Getting the UID of the backup %s needed to be restored", backupName)
```

Comment:
Also , we can remove this as we are getting the UID  in previous step now

Reply:
Done